test/test_save_as_netcdf.py

```diff
@@ -182,22 +188,27 @@ def test_write_with_velocities():
 
     # Add coordinates for face data.
     grid.build_face_coordinates()
-    grid.save_as_netcdf(fname)
 
-    with netCDF4.Dataset(fname) as ds:
-        assert nc_has_variable(ds, 'mesh2')
-        assert nc_has_variable(ds, 'u')
-        assert nc_has_variable(ds, 'v')
-        assert nc_var_has_attr_vals(ds, 'u', {
-            'coordinates': 'mesh2_face_lon mesh2_face_lat',
-            'location': 'face',
-            'mesh': 'mesh2',
-            })
+    fname = 'temp.nc'
+    with chdir(test_files):
+        grid.save_as_netcdf(fname)
+        ds = netCDF4.Dataset(fname)
+        os.remove(fname)
+
+    assert nc_has_variable(ds, 'mesh2')
+    assert nc_has_variable(ds, 'u')
+    assert nc_has_variable(ds, 'v')
+    assert nc_var_has_attr_vals(ds, 'u', {
+        'coordinates': 'mesh2_face_lon mesh2_face_lat',
+        'location': 'face',
+        'mesh': 'mesh2',
+        })
+    ds.close()
```

Comment:
If I understand the fixture right that is not necessary.  Pytest will take care of that.